plugins/module_utils/network/sonic/config/ospf_area/ospf_area.py

```diff
@@ -0,0 +1,1087 @@
+                if area.get("default_cost") and not can_set_cost:
+                    self._module.fail_json(msg="cannot set default cost for area id {area_id} in ".format(area_id=area['area_id']) +
+                                           "vrf {vrf_name} because it is not stub or NSSA area".format(vrf_name=area['vrf_name']))
+
+            for virtual_link in area.get("virtual_links", []):
+                if "authentication" in virtual_link:
+                    if state != "deleted":
+                        # key is always going to either be encrypted or not.
+                        # key_encrypted field only really should exist if key exists so fill in encrypted if missing and defaults to false
+                        if "key" in virtual_link["authentication"] and "key_encrypted" not in virtual_link["authentication"]:
+                            virtual_link["authentication"]["key_encrypted"] = False
+                for md_key in virtual_link.get("message_digest_keys", []):
+                    if state != "deleted":
+                        if "key" not in md_key:
+                            # any state that is adding config cannot have a missing key
+                            self._module.fail_json(msg="area id'd {area_id} for vrf ".format(area_id=area['area_id']) +
+                                                       "{vrf_name} has missing key in message_digest_keys ".format(vrf_name=area['vrf_name']) +
+                                                       "section of virtual link {vlink}".format(vlink=virtual_link['router_id']))
+                        if "key_encrypted" not in md_key:
+                            md_key["key_encrypted"] = False
+        return config
+
+    def format_area_name(self, area_id):
+        """area names in playbook can be single numbers or as four octet numbers, switch works with area names as the latter.
+        make sure things are in octect format by applying formatting where needed"""
+        if area_id.count(".") < 3:
+            area_int = int(area_id)
+            return ".".join([str(area_int >> 24 & 0xff), str(area_int >> 16 & 0xff), str(area_int >> 8 & 0xff), str(area_int & 0xff)])
+        return area_id
+
+    def post_process_diff(self, want, diff, merged_mode=True):
+        '''post process the diff between want and have by keeping any wanted auth key settings together.
+        :param want: the config that the difference is based off of. the first item in the get_diff call
+        :param diff: the diff between want and have config in argspec format. assumes diff is a subset of want'''
+        # whatever values were set for key and key encrypted should be kept together
+        post_cleaned_diff = []
+        for area_w in want:
+            if merged_mode and len(area_w) == 2:
+                # commands has an area with no settings to merge, that doesn't show up in facts because it can break other stuff,
+                # so putting in step to ignore
```

Comment:
it is allowed by CLI, but I've had issues with running playbook specifying deleting an area, then the area still gets reported by the facts module after. Will need checking if rest endpoints cause same behavior

Reply:
This limitation is because of parsing information retrieved through openconfig rest API. Added file showing the output of trying requests on the REST API.
If an area were configured and nothing configured, the endpoint for getting all areas for a Vrf would report it with just the identifier in config, lsdb, and state sections. The problem was getting stuck in a state where GET would report areas that looked like blank ones, but deletions were not working as expected. In this case, DELETE on a specific area errors out, and DELETE on the areas list for the VRF doesn't cause changes. Facts reporting these would lead to config trying to do something about them.
To avoid the error, Facts couldn't report anything that looked blank because the error case and actual blank both have the same sections. Then because Facts can't handle it, it might be less confusing if Config didn't allow it either.
[BlankAreaEdgecase.txt](https://github.com/user-attachments/files/16276200/BlankAreaEdgecase.txt)

Reply:
Xiao and I have discussed this issue in detail. There is a SONiC bug that falsely leaves empty 'area' configuration after configuring 'inter-area-propagation-polices', then deleting them. It is not possible to distinguish between these cases and 'real' empty areas. Since 'empty' areas provide no significant user functionality, I have agreed that we can omit config and facts handling for this type of area for now (until the SONiC bug is fixed sometime after this PR is merged and released.